messaging-client-databus/src/main/java/com/inmobi/messaging/consumer/databus/DatabusConsumer.java

```diff
@@ -125,6 +125,22 @@ protected void initializeConfig(ClientConfig config) throws IOException {
     } else {
       parseClusterNamesAndMigrateCheckpoint(config, rootDirSplits);
     }
+    /*
+    This timer task starts after an initial delay of 30 minutes
+    and runs every 5 minutes. This guy is responsible for identifying new
+    collector output files and initialising partition readers for them
+     */
+    partitionDiscovererAndReaderCreator = new Timer("PartitionDiscovererAndReaderCreator");
+    partitionDiscovererAndReaderCreator.schedule(new TimerTask() {
+      @Override
+      public void run() {
+        try {
+          createPartitionReaders();
+        } catch (IOException e) {
+          e.printStackTrace();
+        }
+      }
+    }, 30 * MINUTE, 5*MINUTE);
```

Comment:
to avoid race condition, a new initdone flag has been introduced, and the timer task will be keeping a watch over it